cmd/restic/lock.go

```diff
@@ -21,17 +21,17 @@ var globalLocks struct {
 	sync.Once
 }
 
-func lockRepo(ctx context.Context, repo restic.Repository) (*restic.Lock, context.Context, error) {
-	return lockRepository(ctx, repo, false)
+func lockRepo(ctx context.Context, repo restic.Repository, gopts GlobalOptions) (*restic.Lock, context.Context, error) {
+	return lockRepository(ctx, repo, false, gopts)
 }
 
-func lockRepoExclusive(ctx context.Context, repo restic.Repository) (*restic.Lock, context.Context, error) {
-	return lockRepository(ctx, repo, true)
+func lockRepoExclusive(ctx context.Context, repo restic.Repository, gopts GlobalOptions) (*restic.Lock, context.Context, error) {
+	return lockRepository(ctx, repo, true, gopts)
 }
 
 // lockRepository wraps the ctx such that it is cancelled when the repository is unlocked
 // cancelling the original context also stops the lock refresh
-func lockRepository(ctx context.Context, repo restic.Repository, exclusive bool) (*restic.Lock, context.Context, error) {
+func lockRepository(ctx context.Context, repo restic.Repository, exclusive bool, gopts GlobalOptions) (*restic.Lock, context.Context, error) {
```

Comment:
when looking at the tests, I actually think that it's better to just manually pass the two necessary attributes of `GlobalOptions` into `lockRepository`. That is please replace `gopts GlobalOptions` with `retryLock time.Duration, json bool`.